qhana_plugin_registry/tasks/plugin_filter.py

```diff
@@ -29,8 +28,48 @@
 
 TASK_LOGGER = get_task_logger(_name)
 
+DEFAULT_BATCH_SIZE = 500
 
-def evaluate_plugin_filter(filter_string: str) -> list[RAMP]:
+
+def get_plugins_from_filter(filter_dict: dict, plugin_mapping: dict) -> set[RAMP]:
+    match filter_dict:
+        case {"and": filter_expr}:
+            return set.intersection(
+                *(get_plugins_from_filter(f, plugin_mapping) for f in filter_expr)
+            )
+        case {"or": filter_expr}:
+            return set.union(
+                *(get_plugins_from_filter(f, plugin_mapping) for f in filter_expr)
+            )
+        case {"not": filter_expr}:
+            return plugin_mapping.keys() - get_plugins_from_filter(
+                filter_expr, plugin_mapping
+            )
+        case {"tag": tag}:
+            has_tag = lambda p, t: any(tag.tag == t for tag in p.tags)
+            return {p_id for p_id, p in plugin_mapping.items() if has_tag(p, tag)}
+        case {"version": version}:
+            specifier_str = re.sub(
+                r"([^\s,])(\s+)", r"\1,\2", version
+            )  # add commas to whitespace
+            try:
+                specifier = SpecifierSet(specifier_str)
+            except InvalidSpecifier:
+                TASK_LOGGER.warning(f"Invalid version specifier: '{version}'")
+                return set()
+            return {
+                p_id
+                for p_id, p in plugin_mapping.items()
+                if Version(p.version) in specifier
+            }
+        case {"name": name}:
+            return {p_id for p_id, p in plugin_mapping.items() if name == p.name}
+        case _:
+            TASK_LOGGER.warning(f"Invalid filter: '{filter_dict}'")
+            return set()
+
+
+def evaluate_plugin_filter(plugin_filter: dict) -> Generator[RAMP, None, None]:
```

Comment:
Prefer the more general Iterator type over the specific Generator type if the intention of the Generator is only to be an iterator.